mapiproxy/libmapistore/mapistore_notification.c

```diff
@@ -3,525 +3,1355 @@
+
+		/* Add the key/value record */
+		rc = memcached_add(mstore_ctx->notification_ctx->memc_ctx, key, strlen(key),
+				   (char *)ndr->data, ndr->offset, 0, 0);
 	}
-	DLIST_ADD_END(mstore_ctx->notifications, new_list, void);
-#endif
+
+	MAPISTORE_RETVAL_IF(rc != MEMCACHED_SUCCESS, ret_to_mapistore(rc), mem_ctx);
+	talloc_free(mem_ctx);
+	return MAPISTORE_SUCCESS;
 }
 
-#if 0
-static struct mapistore_notification_list *mapistore_notification_process_mqueue_notif(TALLOC_CTX *mem_ctx, 
-										       DATA_BLOB data)
-										       
+
+/**
+   \details Get resolver data for a given resolver entry
+
+   \param mem_ctx pointer to the memory context
+   \param mstore_ctx pointer to the mapistore context
+   \param cn the resolver key to lookup
+   \param countp pointer on the number of hosts returned
+   \param hostsp pointer on the list of host to return
+
+   \note calling function is responsible for freeing hostsp
+
+   \return MAPISTORE_SUCCESS on success, otherwise MAPISTORE error
+ */
+_PUBLIC_ enum mapistore_error mapistore_notification_resolver_get(TALLOC_CTX *mem_ctx,
+								  struct mapistore_context *mstore_ctx,
+								  const char *cn, uint32_t *countp,
+								  const char ***hostsp)
 {
-	struct mapistore_notification_list	*nl;
-	struct mapistore_mgmt_command		command;
-	struct ndr_pull				*ndr_pull = NULL;
-
-	ndr_pull = ndr_pull_init_blob(&data, mem_ctx);
-	ndr_pull_mapistore_mgmt_command(ndr_pull, NDR_SCALARS|NDR_BUFFERS, &command);
-
-	/* verbose */
-	{
-		struct ndr_print	*ndr_print;
-		
-		ndr_print = talloc_zero(mem_ctx, struct ndr_print);
-		ndr_print->print = ndr_print_printf_helper;
-		ndr_print->depth = 1;
-		ndr_print_mapistore_mgmt_command(ndr_print, "command", &command);
-		talloc_free(ndr_print);
-	}
+	TALLOC_CTX				*local_mem_ctx;
+	enum mapistore_error			retval;
+	enum ndr_err_code			ndr_err_code;
+	struct ndr_pull				*ndr;
+	struct mapistore_notification_resolver	r;
+	DATA_BLOB				blob;
+	char					*key = NULL;
+	char					*value = NULL;
+	size_t					value_len = 0;
+	memcached_return_t			rc;
+	uint32_t				flags;
+
+	/* Sanity checks */
+	MAPISTORE_RETVAL_IF(!mstore_ctx, MAPISTORE_ERR_NOT_INITIALIZED, NULL);
+	MAPISTORE_RETVAL_IF(!cn, MAPISTORE_ERR_INVALID_PARAMETER, NULL);
+	MAPISTORE_RETVAL_IF(!countp, MAPISTORE_ERR_INVALID_PARAMETER, NULL);
+	MAPISTORE_RETVAL_IF(!hostsp, MAPISTORE_ERR_INVALID_PARAMETER, NULL);
+	MAPISTORE_RETVAL_IF(!mstore_ctx->notification_ctx, MAPISTORE_ERR_NOT_AVAILABLE, NULL);
+	MAPISTORE_RETVAL_IF(!mstore_ctx->notification_ctx->memc_ctx, MAPISTORE_ERR_NOT_AVAILABLE, NULL);
+
+	local_mem_ctx = talloc_new(NULL);
+	MAPISTORE_RETVAL_IF(!local_mem_ctx, MAPISTORE_ERR_NO_MEMORY, NULL);
+
+	retval = mapistore_notification_resolver_set_key(local_mem_ctx, cn, &key);
+	MAPISTORE_RETVAL_IF(retval, retval, local_mem_ctx);
+
+	value = memcached_get(mstore_ctx->notification_ctx->memc_ctx, key, strlen(key), &value_len,
+			      &flags, &rc);
+	talloc_free(key);
```

Comment:
I think this is not really needed because we will free `local_mem_ctx` in any case